rakelib/appeals_api_oauth.rake

```diff
@@ -5,174 +5,209 @@ require 'net/http'
+      find_non_ccg_scopes = ->(scope_list) { scope_list.flatten.uniq.reject { |s| s.start_with? 'system/' } }
+
+      find_scopes = ccg ? find_ccg_scopes : find_non_ccg_scopes
+
+      if api_name.blank?
+        find_scopes.call(AppealsApi::OpenidAuth::DEFAULT_OAUTH_SCOPES.values)
+      else
+        find_scopes.call(
+          {
+            appeals_status: AppealsApi::V1::AppealsController::OAUTH_SCOPES,
+            contestable_issues: AppealsApi::ContestableIssues::V0::ContestableIssuesController::OAUTH_SCOPES,
+            higher_level_reviews: AppealsApi::HigherLevelReviews::V0::HigherLevelReviewsController::OAUTH_SCOPES,
+            legacy_appeals: AppealsApi::LegacyAppeals::V0::LegacyAppealsController::OAUTH_SCOPES,
+            notice_of_disagreements:
+              AppealsApi::NoticeOfDisagreements::V0::NoticeOfDisagreementsController::OAUTH_SCOPES,
+            supplemental_claims: AppealsApi::SupplementalClaims::V0::SupplementalClaimsController::OAUTH_SCOPES
+          }[api_name.to_sym].values
+        )
+      end
+    end
   end
-
-  puts JSON.pretty_generate(JSON.parse(result.body))
-end
-# rubocop:enable Metrics/MethodLength
-
-API_NAMES = %w[appeals_status contestable_issues higher_level_reviews legacy_appeals
-               notice_of_disagreements supplemental_claims].freeze
-
-def api_scopes(api_name = nil)
-  return %w[appeals.read appeals.write] if api_name.blank?
-
-  {
-    appeals_status: AppealsApi::V1::AppealsController::OAUTH_SCOPES,
-    contestable_issues: AppealsApi::ContestableIssues::V0::ContestableIssuesController::OAUTH_SCOPES,
-    higher_level_reviews: AppealsApi::HigherLevelReviews::V0::HigherLevelReviewsController::OAUTH_SCOPES,
-    legacy_appeals: AppealsApi::LegacyAppeals::V0::LegacyAppealsController::OAUTH_SCOPES,
-    notice_of_disagreements: AppealsApi::NoticeOfDisagreements::V0::NoticeOfDisagreementsController::OAUTH_SCOPES,
-    supplemental_claims: AppealsApi::SupplementalClaims::V0::SupplementalClaimsController::OAUTH_SCOPES
-  }[api_name.to_sym].values.flatten.uniq
-end
-
-def abbreviate_snake_case_name(name)
-  name.scan(/(?<=^|_)(\S)/).join
 end
+# rubocop:enable Metrics/ModuleLength
 
 namespace :appeals_api do
   namespace :token do
     desc 'Get a CCG token for all appeals APIs'
     task ccg: :environment do
-      generate_ccg_token
+      AppealsRakeHelpers.generate_ccg_token
     end
 
-    desc 'Validate an OpenID (CCG or Okta) token'
+    desc 'Get an auth code flow token for all appeals APIs'
+    task auth: :environment do
+      AppealsRakeHelpers.generate_auth_code_flow_token
+    end
+
+    desc 'Validate an OpenID (CCG or auth code flow) token'
     task validate: :environment do
-      validate_token
+      AppealsRakeHelpers.validate_token
     end
 
-    API_NAMES.each do |api_name|
-      namespace abbreviate_snake_case_name(api_name).to_sym do
+    AppealsRakeHelpers::API_NAMES.each do |api_name|
+      namespace api_name.scan(/(?<=^|_)(\S)/).join.to_sym do
```

Comment:
Consider keeping a wrapper for this regex. This seems more difficult to parse at a glance than the existing method name.